docs/lsm/multi_version/multi_version.rst

```diff
@@ -0,0 +1,92 @@
+**************
+Multi-version
+**************
+
+The goal of multi-version lsm is to have versioning as a way to introduce changes to service entities while
+supporting the older behaviour and remaining backwards compatible.
+Each service entity will now have versions, which can introduce breaking changes while maintaining support for older
+versions.
+
+Why do we want to use  multi-version LSM and not different services?
+====================================================================
+A service represents a high level desired state. With a new version we want to introduce some changes to a service
+(i.e. add a new option, or change how some minor things are done), but overall, keep the same intent.
+We want to be able to let a service evolve over time while maintaining backwards compatibility and be able to upgrade
+from one version to another without drastically changing the intent of the service.
```

Comment:
I would rephrase this to a very practical `when to use multi-version lsm`  section. I think you want to use when 1. you want to offer multiple api schema versions to the same service or 2. you want to upgrade a service in a way that is not supported by the automated upgrade mechanism (if possible, you can link to the docs that describe what is supported, if we have such docs)